### PR TITLE
[Feature] add support for str_to_jodatime function

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #26422 

parse and format with Joda format:
```sql
-- format
select jodatime_format(now(), 'MMMM dd yyyy HH:mm:ss');

-- parse
select str_to_jodatime('2023-08-02 14:37:02', 'yyyy-MM-dd HH:mm:ss');
```

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Youngwb** [comment]
plz add test like ' PARSE_DATETIME('2023-05','yyyy-MM')', and I think str_to_jodatime is better

**Comment by wangsimo0** [comment]
It's better rename to str_to_jodatime. tell user this function use joda's standard directly.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 297 / 319 (93.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/time_functions.cpp|27|33|81.82%|[1720, 1730, 1733, 1791, 1812, 1813]|
|:large_blue_circle:|src/runtime/datetime_value.cpp|270|286|94.41%|[173, 174, 215, 282, 292, 331, 332, 355, 356, 359, 360, 461, 462, 463, 489, 490]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/time_functions.cpp [hunk 1]
```diff
@@ -1704,15 +1651,65 @@ StatusOr<ColumnPtr> TimeFunctions::str_to_date_uncommon(FunctionContext* context
...
+StatusOr<ColumnPtr> TimeFunctions::parse_datetime(FunctionContext* context, const Columns& columns) {
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    size_t size = columns[0]->size(); // minimum number of rows.
+    std::string_view format_str = ColumnHelper::get_const_value<TYPE_VARCHAR>(columns[1]);
+    ColumnBuilder<TYPE_DATETIME> result(size);
+    auto str_viewer = ColumnViewer<TYPE_VARCHAR>(columns[0]);
+
+    joda::JodaFormat formatter;
+    if (!formatter.prepare(format_str)) {
```
[the remark this answers is not in the document]
Comment (mofeiatwork) [reply be/src/exprs/time_functions.cpp]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 237 / 346 (68.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/datetime_value.h|0|2|00.00%|[536, 537]|
|:large_blue_circle:|src/exprs/time_functions.cpp|0|45|00.00%|[1730, 1731, 1732, 1735, 1736, 1782, 1783, 1784, 1785, 1787, 1790, 1791, 1792, 1795, 1796, 1798, 1799, 1800, 1801, 1803, 1806, 1807, 1808, 1809, 1812, 1813, 1815, 1816, 1817, 1819, 1820, 1821, 1823, 1824, 1825, 1827, 1829, 1830, 1831, 1833, 1835, 1836, 1841, 1850, 1852]|
|:large_blue_circle:|src/runtime/datetime_value.cpp|237|299|79.26%|[141, 144, 150, 154, 162, 163, 164, 165, 166, 167, 169, 170, 172, 173, 175, 176, 177, 178, 179, 180, 188, 199, 217, 229, 230, 231, 232, 233, 234, 236, 237, 238, 239, 240, 241, 250, 262, 273, 279, 287, 299, 317, 320, 323, 337, 340, 343, 358, 371, 384, 398, 409, 428, 429, 430, 438, 445, 450, 459, 469, 478, 1185]|

**Review by satanson — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0, branch-3,1 branch-3.2

**Review by stdpain — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0, branch-3.1, branch-3.2

**Pull request merged**